compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -219,6 +219,87 @@ void addCPUBufferOpsTileAndVectorizePipeline(OpPassManager &passManager) {
   }
 }
 
+void addDoubleTilingPadExpertPassPipeline(OpPassManager &passManager) {
+  passManager.addPass(createVerifyLinalgTransformLegalityPass());
+  addTileAndDistributePasses(passManager, /*convertToDestinatinoStyle=*/true);
+
+  {
+    LinalgFusePassOptions options;
+    options.tilingLevel =
+        static_cast<int64_t>(StrategyTilingLevel::ParallelTiles);
+    passManager.addNestedPass<func::FuncOp>(createLinalgFusePass(options));
+    passManager.addNestedPass<func::FuncOp>(createCanonicalizerPass());
+    passManager.addNestedPass<func::FuncOp>(createCSEPass());
+  }
+
+  auto pad = [&](std::string anchorOpName, ArrayRef<int64_t> paddingDims) {
+    LinalgFusePassOptions options;
+    options.pad = true;
+    options.anchorOpName = anchorOpName;
+    options.paddingDimensions.assign(paddingDims.begin(), paddingDims.end());
+    passManager.addNestedPass<func::FuncOp>(createLinalgFusePass(options));
+  };
+
+  SmallVector<int64_t> paddingDims = {0, 1};
+  pad("linalg.fill", paddingDims);
```

Comment:
I missed this on review. Is there a reason we have to call out operations this way. Could we just do it for all the `computeOps` that are found?

Reply:
The order matters. This falls to the problem of having better control for scheduling. Ideally, the order is:

1. Tile parallel loops
2. Pad operations in top-down order (i.e., topology order) for all parallel  dims.
3. Tile reduction loops
4. Pad operations for all reduction dims. (In IREE's use case, there is only one operation that has reduction loops.)

I may be able to wrap them into a new pass. I'm not sure how tricky it would be, but it's the thing we want to support eventually.

I just found that the padding approach only works for `linalg.matmul` ops. Other ops like `linalg.batch_matmul` may not work because they have more parallel dims and they are not padded correctly. I'll try to fix that or only enable it for matmul cases.